src/plugins/data/server/kql_telemetry/usage_collector/fetch.test.ts

```diff
@@ -89,40 +93,64 @@ describe('makeKQLUsageCollector', () => {
 
     it('should return opt in data from the .kibana/kql-telemetry doc', async () => {
       setupMockCallCluster({ optInCount: 1 }, 'kuery');
-      const fetchResponse = await fetch(callCluster);
+      collectorFetchClients = {
+        callCluster,
+        esClient: esClientMock,
+      };
```

Comment:
I think it's best to define it as:

```suggestion
      collectorFetchClients = {
        ...collectorFetchClientsMock,
        callCluster,
      };
```

With `const collectorFetchClientsMock = createCollectorFetchClientsMock();` as a global variable, defined at the top.

Or:

```suggestion
      const collectorFetchClients = createCollectorFetchClientsMock();
      collectorFetchClients.callCluster = callCluster;
```

This way, when we extend the provided clients, we won't need to revisit these tests.